tests/unit/core/utils/test_index.js

```diff
@@ -886,4 +888,38 @@ describe(__filename, () => {
       ).toEqual(undefined);
     });
   });
+
+  describe('getLocalizedTextWithLinkParts', () => {
+    it('returns a descriptive object', () => {
+      const parts = getLocalizedTextWithLinkParts({
+        i18n: fakeI18n(),
+        text: 'Explore more %(linkStart)s stuff %(linkEnd)s here.',
+      });
+
+      expect(parts).toHaveProperty('beforeLinkText');
+      expect(parts.beforeLinkText).toEqual('Explore more ');
+
+      expect(parts).toHaveProperty('innerLinkText');
+      expect(parts.innerLinkText).toEqual(' stuff ');
+
+      expect(parts).toHaveProperty('afterLinkText');
+      expect(parts.afterLinkText).toEqual(' here.');
+    });
+
+    it('only returns the beforeLinkText when the linkStart or linkEnd are not found', () => {
```

Comment:
Why do we want to support this case? It seems like we should throw an error if `linkStart` or `linkEnd` are not found because isn't the point of this to be used for a link?

Reply:
I guess not :/ I will remove, thanks :)

Reply:
I guess the best thing would be to throw an Error